### PR TITLE
ARROW-6365: [R] Should be able to coerce numeric to integer with schema

### PR DESCRIPTION
**Pull request by romainfrancois**
``` r
arrow::array(10, type = arrow::int32())
#> arrow::Array 
#> [
#>   10
#> ]
```

<sup>Created on 2019-09-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

### CODE REVIEW

**Comment by nealrichardson** [comment]
There's a C++ lint failure: https://travis-ci.org/apache/arrow/jobs/582564448#L690

But otherwise LGTM.

File: r/tests/testthat/test-Array.R [hunk 1]
```diff
@@ -387,13 +387,20 @@ test_that("array() aborts on overflow", {
...
   for(type in types) {
-    expect_error(array(10, type = type)$type, "Cannot convert.*REALSXP")
+    a <- array(10, type = type)
+    expect_equal(a$type, type)
+
+    # exception for now because we cannot handle
+    # unsigned 64 bit integers yet
+    if (type != uint64()) {
+      expect_true(a$as_vector() == 10L)
+    }
```
Comment (wesm) [inline r/tests/testthat/test-Array.R]: If the values are not whole numbers, do they get truncated?

Comment (nealrichardson) [reply r/tests/testthat/test-Array.R]: It appears to truncate: 

```r
> array(c(10, 10.4, 10.7, 11.8), int32())
arrow::Array 
[
  10,
  10,
  10,
  11
] 
```

**Comment by nealrichardson** [comment]
Lint passed on romain's fork: https://travis-ci.org/romainfrancois/arrow/jobs/582792569

So if there are no objections, I'll merge this.

File: r/tests/testthat/test-Array.R (hunk 1 above)
Comment (romainfrancois) [reply r/tests/testthat/test-Array.R]: Truncation comes from the `static_cast()` here: https://github.com/romainfrancois/arrow/blob/ed12b6ecf85303072e605d676679a27e7a33edd7/r/src/array_from_vector.cpp#L224

**Pull request closed without merging**